src/lib/WIFI/devWIFI.cpp

```diff
@@ -796,6 +801,238 @@ static void WebUploadForceUpdateHandler(AsyncWebServerRequest *request) {
   }
 }
 
+#if defined(RADIO_LR1121)
+static size_t expectedFilesize;
+static SX12XX_Radio_Number_t updatingRadio;
+extern LR1121Hal hal;
+static uint8_t packet[256];
```

Comment:
Rather than having this hanging around in RAM all the time for the unlikely event that someone is going to flash their LR1121 chips, why not make this a dynamically malloced update struct that contains all of this state information?
```cpp
struct lr1121UpdateState_s {
  SX12XX_Radio_Number_t updatingRadio;
  size_t left_over;
  struct {
    uint8_t header[6];
    uint8_t buffer[256];
  } PACKED packet;
};
static lr1121UpdateState_s *lr1121UpdateState;
```

And also note how the header and buffer combined, so that later on you don't need to allocate 6+256 off the stack to make another copy.